app_dart/test/service/branch_service_test.dart

```diff
@@ -53,6 +54,22 @@ void main() {
       when(githubService.github.repositories).thenReturn(mockRepositoriesService);
     });
 
+    test('return 404 when branch version file does not exist', () async {
+      final gh.Branch candidateBranch = generateBranch(3, name: 'flutter-3.4-candidate.5', sha: '789dev');
+      when(mockRepositoriesService.listBranches(any)).thenAnswer((Invocation invocation) {
+        return Stream.fromIterable([candidateBranch]);
+      });
+      when(mockRepositoriesService.getContents(any, any)).thenAnswer((Invocation invocation) {
+        throw Exception('404 file not found');
+      });
+      final List<Map<String, String>> result =
+          await branchService.getReleaseBranches(githubService: githubService, slug: Config.cocoonSlug);
+      final betaBranch = result.singleWhere((Map<String, String> branch) => branch['name'] == 'beta');
+      expect(betaBranch['branch'], '');
```

Comment:
Writing tests using behavior driven development makes them easier to follow

```suggestion
      expect(betaBranch['branch'].isEmpty, isTrue);
```

Reply:
Sure, updated. 

umm but I thought returning an empty string might be the behavior we are expecting?

Reply:
`''` is harder to see than `isEmpty`. In BDD, making tests read like a sentence is part of the goal.